src/lib/y2packager/medium_type.rb

```diff
@@ -0,0 +1,107 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2019 SUSE LLC, All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+# ------------------------------------------------------------------------------
+
+require "yast"
+
+require "y2packager/product_location"
+require "y2packager/repomd_downloader"
+
+Yast.import "InstURL"
+
+module Y2Packager
+  # This class scans the installation medium type and detects which installation
+  # medium type it is (online/offline/standard).
+  class MediumType
+    extend Yast::Logger
```

Comment:
hmm, whole class is class based, so what about simple `class << self` and then common include and also private?